pylint/test/functional/mutable_sequence_modified_in_loop.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+""" The mutable sequence was modified inside the loop """
+
+# pylint:disable=print-statement, missing-docstring, redefined-outer-name, invalid-name, using-constant-test, no-member
+
+
+class MyList(object):
+
+    def __init__(self):
+        self.my_list = range(1, 6)
+
+    def __iter__(self):
+        return iter(self.my_list)
+
+    def remove(self, item):
+        pass
+
+    def delete(self, item):
+        pass
+
+    def pop(self, item):
+        pass
+
+
+my_list = MyList()
+for item in my_list:
+    my_list.remove(item)
+
+my_list = MyList()
+for item in my_list:
+    my_list.delete(item)
+
+my_list = MyList()
+for item in my_list:
+    my_list.pop(item)
+
+
+items = range(1, 6)
```

Comment:
Add a line `from missing_module import uninferable_object` and check a case when you iterating over `uninferable_object`. It should not emit an error because you don't know what `uninferable_object` is. Your code will emit that error there, which will be a false positive (at least potentially).